app/src/features/charts/MapNavChart.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from "react";
+import UsaChloroplethMap from "../charts/UsaChloroplethMap";
+import { USA_FIPS, USA_DISPLAY_NAME, STATE_FIPS_MAP } from "../../utils/Fips";
+import Alert from "@material-ui/lab/Alert";
+import Breadcrumbs from "@material-ui/core/Breadcrumbs";
+import Link from "@material-ui/core/Link";
+import Typography from "@material-ui/core/Typography";
+import { VariableId } from "../../utils/variableProviders";
+
+function GeographyBreadcrumb(props: {
+  text: string;
+  isClickable: boolean;
+  onClick?: () => void;
+}) {
+  return (
+    <>
+      {props.isClickable && (
+        <Link color="inherit" onClick={() => props.onClick!()}>
+          {props.text}
+        </Link>
+      )}
+      {!props.isClickable && (
+        <Typography color="textPrimary">{props.text}</Typography>
+      )}
+    </>
+  );
+}
+
+function MapNavChart(props: {
+  varField: VariableId;
+  varFieldDisplayName: string;
+  fipsGeo: string;
+  countyFips: string | undefined;
+  data: Record<string, any>[];
+  updateGeoCallback: (message: string) => void;
+}) {
+  const [countyName, setCountyName] = useState<string>();
+
+  useEffect(() => {
+    setCountyName(undefined);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [props.fipsGeo]);
+
+  const signalListeners: any = {
+    click: (...args: any) => {
+      const clickedData = args[1];
+      props.updateGeoCallback(clickedData.id);
+      if (clickedData.id.length === 5) {
+        setCountyName(clickedData.properties.name);
+      }
+    },
+  };
+
+  // TODO - make the mouse turn into a pointer when you hover over
+  return (
+    <div>
+      {props.fipsGeo !== USA_FIPS && (
+        <Alert severity="error">
+          This dataset does not provide county level data
+        </Alert>
+      )}
+      <Breadcrumbs aria-label="breadcrumb">
+        <GeographyBreadcrumb
+          text={USA_DISPLAY_NAME}
+          isClickable={props.fipsGeo !== USA_FIPS}
+          onClick={() => {
+            props.updateGeoCallback(USA_FIPS);
+            setCountyName(undefined);
+          }}
+        />
+        {props.fipsGeo !== USA_FIPS && (
+          <GeographyBreadcrumb
+            text={STATE_FIPS_MAP[props.fipsGeo]}
+            isClickable={!!countyName}
+            onClick={() => {
+              props.updateGeoCallback(props.fipsGeo.substring(0, 2));
```

Comment:
I like creating utility functions for this kind of thing like `getStateFips(countyFips)` so it's more obvious what's happening, easier to refactor, and harder to mistype.

As an improvement, I think it could be nice to create a class for FIPS, which improves typechecking and allows these utility methods to be instance methods. I wouldn't convert any of the datasets, I'd just make it something you create on the fly from a fips code, and then pass into components:

```
const fips = new Fips('10321');
const state = fips.getStateFips();
const isCounty = fips.isCounty();
...
```

Reply:
Ya this is a great suggestion I think this makes a lot of things more clear... I think this one change addressed some other concerns you raised as well